src/utils/fireDestinations.js

```diff
@@ -29,73 +28,46 @@ export default ({ logger, destinations }) => {
     return iframePromise;
   };
 
-  const fireInIframe = ({ attributes }) => {
+  const fireInIframe = ({ src }) => {
     return createIframe().then(iframe => {
       const currentDocument = iframe.contentWindow.document;
-      fireImage({ attributes, currentDocument });
+      return fireImage({ src, currentDocument });
     });
   };
 
   return Promise.all(
     destinations.map(dest => {
-      return new Promise(resolve => {
-        if (isNonEmptyString(dest.url)) {
-          if (!isNil(dest.hideReferrer)) {
-            const attributes = {
-              onload: () => {
-                resolve({
-                  status: "loaded",
-                  dest
-                });
-              },
-              onerror: () => {
-                logger.log(`Destination failed: ${dest.url}`);
-                resolve({
-                  status: "errored",
-                  dest
-                });
-              },
-              onabort: () => {
-                logger.log(`Destination aborted: ${dest.url}`);
-                resolve({
-                  status: "aborted",
-                  dest
-                });
-              },
-              src: dest.url
-            };
+      const imagePromise = dest.hideReferrer
+        ? fireInIframe({ src: dest.url })
+        : fireOnPage({ src: dest.url });
 
-            if (dest.hideReferrer) {
-              fireInIframe({ attributes });
-            } else {
-              fireOnPage({ attributes });
-            }
-          } else {
-            logger.error(
-              `Destination hideReferrer property is not defined for url ${
-                dest.url
-              } .`
-            );
-          }
-        } else {
-          logger.error("Destination url is not a populated string.");
-        }
```

Comment:
If we don't end up merging this PR, we need to add this code to each of the `else` statements:

```js
resolve({	
  status: "errored",	
  dest	
});
```

Otherwise, an error will be thrown down on line 92 when we try to filter results.